### PR TITLE
[FEATURE] extend item table with currency-change functionallity

### PR DESCRIPTION
**Pull request by michaelzangerle**
**Requires:**
- https://github.com/sulu-cmf/SuluProductBundle/pull/67

**Tasks:**
- [x] done
- [ ] got feedback

**Informations:**

| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| Tests pass? | none |
| Fixed tickets | none |
| Doc | none |

### CODE REVIEW

File: Resources/public/js/components/item-table/main.js [hunk 1]
```diff
@@ -55,8 +55,9 @@ define([
         },
 
         constants = {
-            formId: '#item-table-form',
+            formId: '#item-table-form', // FIXME should be changed to a unique id when multiple instances are on one page
```
Comment (turbo-ele) [inline Resources/public/js/components/item-table/main.js:58]: actually, this is already the case. we should change that :)

File: Resources/public/js/components/item-table/main.js [hunk 2]
```diff
@@ -133,13 +131,21 @@ define([
          * @param key
          * @param value
          *
-         * @event sulu.item-table[.INSTANCENAME].set-default-data
+         * @event sulu.item-table.[INSTANCENAME].set-default-data
```
Comment (turbo-ele) [inline Resources/public/js/components/item-table/main.js]: actually that would mean, that if instancename is not set the event would be 

```
sulu.item-table..set-default-data
```

File: Resources/public/js/components/item-table/main.js [hunk 3]
```diff
@@ -200,6 +207,154 @@ define([
...
+                // go on when loader is fully loaded and product data retrieved
+                // dfdLoadedProducts needs to be the first param
+                this.sandbox.dom.when(dfdLoadedProducts, dfdLoaderStarted)
+                    .done(function(data) {
+                        updatePricesForEachProduct.call(this, data);
+                        updateGlobalPrice.call(this);
+                        stopLoader.call(this);
+                    }.bind(this))
+                    .fail(function(jqXHR, textStatus, error) {
+                        this.sandbox.logger.error(jqXHR, textStatus, error);
```
Comment (turbo-ele) [inline Resources/public/js/components/item-table/main.js:242]: we should show an error label in that case

File: Resources/public/js/components/item-table/main.js [hunk 4]
```diff
@@ -200,6 +207,154 @@ define([
...
+
+        /**
+         * Updates the price for every product and the total
+         * @param data
+         */
+        updatePricesForEachProduct = function(data) {
+            var $el,
+                item,
+                prop,
+                prices = getArrayForProductPrices.call(this, data._embedded.products);
```
Comment (turbo-ele) [inline Resources/public/js/components/item-table/main.js:256]: if we'd fetch products with a collection, then 

```
data._embedded.products
```

could be put into the parse function

File: Resources/public/js/components/item-table/main.js [hunk 5]
```diff
@@ -200,6 +207,154 @@ define([
...
+
+                    // get price
+                    if (!!prices[item.product.id] && !!prices[item.product.id][this.currency]) {
+                        item.price = prices[item.product.id][this.currency];
+                    } else {
+                        item.price = 0;
+                    }
+
+                    // update input in dom
+                    $el = this.sandbox.dom.find('.price', this.sandbox.dom.find('#' + prop, this.$list));
```
Comment (turbo-ele) [inline Resources/public/js/components/item-table/main.js]: use constants

File: Resources/public/js/components/item-table/main.js [hunk 6]
```diff
@@ -200,6 +207,154 @@ define([
...
+                }.bind(this));
+        },
+
+        /**
+         * Creats dom element for loader and appends it to
+         */
+        prepareDomForLoader = function() {
+            var height = this.sandbox.dom.height(this.$el);
+
+            this.$loader = this.sandbox.dom.createElement('<div style="display:hidden" class="grid-row ' + constants.loaderClass + '"></div>');
```
Comment (turbo-ele) [inline Resources/public/js/components/item-table/main.js]: line length. what about put this code into template and just add a div each time?

**Comment by turbo-ele** [comment]
ack

File: Resources/public/js/components/item-table/main.js (hunk 1 above)
Comment (michaelzangerle) [inline Resources/public/js/components/item-table/main.js:58]: Added Issue: https://github.com/massiveart/POOL-ALPIN/issues/425

**Pull request merged**